http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpDockerDAO.scala

```diff
@@ -178,12 +191,25 @@ object HttpDockerDAO {
       digest <- cursor.get[String]("digest")
     } yield ManifestConfig(mediaType, size, digest)
   }
-  implicit val containerConfigDecoder: Decoder[ContainerConfig] = Decoder.instance { d =>
-    val cursor = d.downField("container_config")
-    for {
-      image <- cursor.get[String]("Image")
-      env <- cursor.get[List[String]]("Env")
-    } yield ContainerConfig(image, env)
+
+  implicit val envDecoder: Decoder[ContainerEnv] = {
+    lazy val containerConfigDecoder: Decoder[ContainerEnv] = Decoder.instance { d =>
+      val cursor = d.downField("container_config")
+
+      for {
+        env <- cursor.get[List[String]]("Env")
+      } yield ContainerEnv(env)
+    }
+
+    lazy val configDecoder: Decoder[ContainerEnv] = Decoder.instance { d =>
+      val cursor = d.downField("config")
+
+      for {
+        env <- cursor.get[List[String]]("Env")
+      } yield ContainerEnv(env)
+    }
+
+    containerConfigDecoder or configDecoder
```

Comment:
Circe selects the first decoder that succeeds.

Reply:
I think decoder can be 
```
Decoder.instance { d =>
      for {
        envOpt <- d.downField("container_config").as[Option[List[String]]]
        env <- envOpt.fold(d.downField("config").as[List[String]])(ls => ls.asRight[DecodingFailure])
      } yield ContainerEnv(env)
    }
```

Reply:
That wouldn't work because the final field we're looking for is `Env` which may be under `container_config` or `config`. It wasn't clear to me how the for-comprehension in your message can be modified to accommodate for that and since the PR'ed version is easy to read IMO, I'm inclined to not change it. What do you think?

Reply:
I think the current decoder is a bit repetitive..but if you prefer that, I won't insist